st2reactor/st2reactor/garbage_collector/base.py

```diff
@@ -205,6 +224,54 @@ def _purge_action_executions(self):
 
         return True
 
+    def _purge_workflow_executions(self):
+        """
+        Purge workflow executions and corresponding live action, stdout and stderr object which match
+        the criteria defined in the config.
+        """
+        utc_now = get_datetime_utc_now()
+        timestamp = (utc_now - datetime.timedelta(days=self._action_executions_ttl))
+
+        # Another sanity check to make sure we don't delete new executions
+        if timestamp > (utc_now - datetime.timedelta(days=MINIMUM_TTL_DAYS)):
+            raise ValueError('Calculated timestamp would violate the minimum TTL constraint')
+
+        timestamp_str = isotime.format(dt=timestamp)
+        LOG.info('Deleting workflow executions older than: %s' % (timestamp_str))
+
+        assert timestamp < utc_now
+
+        try:
+            purge_workflow_execution(logger=LOG, timestamp=timestamp)
+        except Exception as e:
+            LOG.exception('Failed to delete workflow executions: %s' % (six.text_type(e)))
+
+        return True
+
+    def _purge_task_executions(self):
+        """
+        Purge task executions and corresponding live action, stdout and stderr object which match
+        the criteria defined in the config.
+        """
+        utc_now = get_datetime_utc_now()
+        timestamp = (utc_now - datetime.timedelta(days=self._action_executions_ttl))
```

Comment:
Think this should be _task_execution_ttl instead of actions

Reply:
> Think this should be _task_execution_ttl instead of actions

Thank you @amanda11 will make this change.